selfdrive/car/docs_definitions.py

```diff
@@ -30,120 +30,144 @@ class Star(Enum):
   EMPTY = "empty"
 
 
-class PartType(Enum):
-  connector = "Connector"
-  device = "Device"
-  cable = "Cable"
-  accessory = "Accessory"
-  mount = "Mount"
-
-
+# A part + its comprised parts
 @dataclass
-class Part:
+class BasePart:
   name: str
+  # A list of parts that come with this part in the shop
+  parts: List[Enum] = field(default_factory=list)
+  # If there are extra parts included, you can override required_parts
+  required_parts: Optional[List[Enum]] = None
```

Comment:
I thought we decided to keep everything extra/non-required in web3, @adeebshihadeh have any preference for required parts vs. what the shop includes?

Reply:
yeah extra stuff shop includes should go in web3

Reply:
@maxime-desroches let's remove the required stuff then. I added to bring everything into openpilot to see what it would look like, but it makes more sense in web3